packages/core/src/RenderPipeline/BatchUtils.ts

```diff
@@ -0,0 +1,62 @@
+import { SpriteMask, SpriteMaskInteraction, SpriteRenderer } from "../2d";
+import { SubRenderElement } from "./SubRenderElement";
+
+/**
+ * @internal
+ */
+export class BatchUtils {
+  static canBatchSprite(elementA: SubRenderElement, elementB: SubRenderElement): boolean {
+    if (elementA.subChunk.chunk !== elementB.subChunk.chunk) {
+      return false;
+    }
+
+    const rendererA = <SpriteRenderer>elementA.component;
+    const rendererB = <SpriteRenderer>elementB.component;
+    const maskInteractionA = rendererA.maskInteraction;
+
+    // Compare mask, texture and material
+    return (
+      maskInteractionA === rendererB.maskInteraction &&
+      (maskInteractionA === SpriteMaskInteraction.None || rendererA.maskLayer === rendererB.maskLayer) &&
+      elementA.texture === elementB.texture &&
+      elementA.material === elementB.material
```

Comment:
elementA.material.shader === elementB.material.shader && elementA.material.shader.canBatch(renderer, material);

Reply:
> why not try:
> 
> ```
> elementA.material.shader === elementB.material.shader && elementA.material.shader.canBatch(renderer, material);
> ```

all sprite used the same shader, but diffrent texture, texture is belong to renderer